classes/class-wc-connect-nux.php

```diff
@@ -83,5 +94,168 @@ public function register_add_service_to_zone_pointer( $pointers ) {
 			);
 			return $pointers;
 		}
+
+		public function get_jetpack_install_status() {
+			$status = Jetpack_Install_Status::UNINSTALLED;
+
+			if( 0 === validate_plugin( 'jetpack/jetpack.php' ) ) {
+				$status = Jetpack_Install_Status::INSTALLED;
+			}
+
+			if ( ! class_exists( 'Jetpack_Data' ) ) {
+				// installed, but not activated
+				return $status;
+			}
+
+			if ( defined( 'JETPACK_DEV_DEBUG' ) ) {
+				$status = Jetpack_Install_Status::DEV;
+				return $status;
+			}
+
+			if ( ! method_exists( 'Jetpack_Data', 'get_access_token' )
+				|| ! class_exists( 'Jetpack' )
+			) {
+				$status = Jetpack_Install_Status::ERROR;
+				return $status;
+			}
+			$status = Jetpack_Install_Status::ACTIVATED;
+
+
+			$user_token = Jetpack_Data::get_access_token( JETPACK_MASTER_USER );
+			if ( $user_token
+				&& is_object( $user_token )
```

Comment:
Neat! `$user_token` can be false, but that works for `isset()`, too. TIL 😄 Now we just have the one check: `$user_token->external_user_id` 👌